py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py

```diff
@@ -36,29 +42,176 @@ def slice_op(  # TODO: This should be slice not whatever is in base
     stop: Optional[int],
     step: int,
 ) -> TRTTensor:
+    # check if dim is same as dynamic shape dimension
+    # this is required when stop is ITensor
+    dynamic_input_dim_equal = False
+    for i in range(len(input.shape)):
+        if input.shape[i] == DYNAMIC_DIM and i == dim:
+            dynamic_input_dim_equal = True
+
     # Special case for start being None
     if start is None:
         start = 0
 
     # Special case for stop being None
+    stop_dynamic_None = False
+    if stop is None:
+        stop_dynamic_None = True if input.shape[dim] == -1 else False
     if stop is None:
-        stop = input.shape[dim]
+        stop = 0 if input.shape[dim] == -1 else input.shape[dim]
 
     dim = get_positive_dim(dim, len(input.shape))
-    start = get_positive_dim(start, input.shape[dim])
-    stop = get_positive_dim(stop, input.shape[dim])
 
-    if has_dynamic_shape(input.shape):
-        # Check whether slice target dim is dynamic shape dim
-        assert input.shape[dim] != -1, "Can't slice on dynamic shape dimension!"
+    # Assign the initial start tensor
+    start_slice = []
+    # the add_slice will take care of dynamic input shape cases here
+    if isinstance(start, int):
+        start_slice = [0] * len(input.shape)
+        start_slice[dim] = start
+    else:
+        for i in range(len(input.shape)):
+            start_slice.append(0) if i == dim else start_slice.append(start)
```

Comment:
Can you explain what is happening here ? Why are we appending start (an ITensor) to all the other dimensions except dim ?

Reply:
Thanks for pointing this out. start_slice should have start appended for i==dim.